src/UserSpaceInstrumentation/Trampoline.cpp

```diff
@@ -202,8 +207,13 @@ void AppendCallToEntryPayloadAndOverwriteReturnAddress(uint64_t entry_payload_fu
   trampoline.AppendBytes({0x48, 0x83, 0xc0, 0x40})
       .AppendBytes({0x50})
       .AppendBytes({0x48, 0x8b, 0x38})
-      .AppendBytes({0x48, 0xbe})
-      .AppendImmediate64(function_address)
+      .AppendBytes({0x48, 0xbe});
+  // This fails if the code for the trampoline was changed - see the comment at the declaration of
+  // kOffsetOfFunctiunIdInCallToEntryPayload above.
+  CHECK(trampoline.GetResultAsVector().size() == kOffsetOfFunctiunIdInCallToEntryPayload);
+  // The value of function id will be overwritten by every call to `InstrumentFunction`. The zero
+  // here is just a placeholder.
+  trampoline.AppendImmediate64(0)
```

Comment:
or maybe use something big and telling here instead of 0. 0xDEADBEEF?

Reply:
There is even space for two dead cows.